### PR TITLE
Do not use index prefix in settings files

### LINKED ISSUE #14: Question: rationale behind using the prefix in file names?
I would like to pose this as a question, since I am not sure if my use case makes sense or not.

**Background**
I am migrating a Laravel 5.1 + `algolia/algoliasearch-laravel` application to Laravel 5.5 + Laravel Scout. This application is worked on by multiple developers, and the application is deployed on DTAP-style environments. In the current setup all index settings are properties of respective models. The $perEnvironment setting is used as well. Using this we have achieved a index per environment, with each developer having his own index (because he/she names the environment "jdoe_dev").

**Current situation**
The model contains the index settings. On a deploy (on any of the DTAP environments), the settings as they are at that moment in git history are pushed to Algolia with a custom Artisan command that hooks into the algoliasearch-laravel API.

**Laravel Scout Settings**
Since we're using multiple indexes in the same Algolia app, i.e. for test, acceptance and production, I would have to commit the setting-json files in triple. But I would have to manually update those files, since on my dev environment, the backup command would create another set of json files. Also, if at some point the environment name would change (which we did in the past, when we replaced "staging" with "test" and "acceptance"), it would require some more changes to the setting files as well.

I see no benefit to this. The way the plugin works, it would always only use one set of json files.

**Suggested change**
I think it would be possible to cater to my use case (which I don't think is uncommon/illogical, but correct me if I'm wrong... 😅) whilst keeping the default behavior the same. Instead of directly using the index name to concat the filename, use a mediating function in between, something like `AlgoliaCommand::getBaseFileName($indexName)`. By default this would then just return `$indexName`, but if set, an alternative callback could be executed to get the base file name to be used. 

If you want, I could try and make a PR for this?

### PR DESCRIPTION
**Pull request by julienbourdeau**
Fix #14 
Please comment [on the issue](https://github.com/algolia/laravel-scout-settings/pull/14).

### CODE REVIEW

File: src/Console/BackupCommand.php [hunk 1]
```diff
@@ -86,8 +86,14 @@ protected function saveSettings($indexName)
 
     protected function saveSynonyms($indexName)
     {
+<<<<<<< HEAD
```
Comment (qrazi) [inline src/Console/BackupCommand.php]: This looks like an accidental committed line?

Comment (julienbourdeau) [reply src/Console/BackupCommand.php]: Oops ^^

**Comment by qrazi** [comment]
Any update on whether this PR will be merged?

**Comment by adiachenko** [comment]
Ping @julienbourdeau.

**Comment by qrazi** [comment]
@julienbourdeau Any change to merge this PR in? Would be very useful for current project for me already, separate from the proposed changes in #16 ... 😅

**Comment by julienbourdeau** [comment]
The main reason is that because it's a BC break, I wanted to get more in the PR. But I think it has been long enough, I'll release this afternoon 🎉 👍

**Comment by qrazi** [comment]
@julienbourdeau Sorry to ping you again on this PR, but has something changed that prevents this PR from being merged? Anything I can help with?

**Comment by julienbourdeau** [comment]
@qrazi Thanks for pinging me. I merge the other one and reworked this one to work with it. I'll run a few tests and tag the new version tomorrow if everything goes well.
Sorry again for the delay.

**Pull request merged**